apps/registry/codemods/react-router/4/replace-location-query/src/index.ts

```diff
@@ -0,0 +1,44 @@
+import type { API, FileInfo } from 'jscodeshift';
+
+function transform(file: FileInfo, api: API) {
+	const j = api.jscodeshift;
+	const root = j(file.source);
+
+	root.find(j.MemberExpression, {
+		object: {
+			name: 'location',
+		},
+		property: {
+			name: 'query',
+		},
+	}).replaceWith(() =>
+		j.callExpression(j.identifier('parse'), [
+			j.memberExpression(
+				j.identifier('location'),
+				j.identifier('search'),
+			),
+		]),
+	);
+
+	const hasQueryStringImport =
+		root.find(j.ImportDeclaration, {
+			source: {
+				value: 'query-string',
+			},
+		}).length > 0;
+
+	if (!hasQueryStringImport) {
```

Comment:
will this codemod insert `query-string` import even if we did not replace `location.query`? I think we need to add a `dirtyFlag` and insert import only when we replaced location.query

Reply:
addressed and added a test